eggo/dag.py

```diff
@@ -140,119 +155,91 @@ def output(self):
-                    else:
-                        compression_ext = os.path.splitext(source['url'])[-1]
-                        if compression_ext == '.gz':
-                            compression_type = 'GZIP'
-                        else:
-                            raise ValueError("Unknown compression type: {0}".format(
-                                compression_ext))
-                    command_file.write(('{ephem} {source} {compress} {tmp_s3_path} '
-                                        '{final_path}\n').format(
-                                      ephem=EPHEMERAL_MOUNT, source=source['url'],
-                                      compress=compression_type,
-                                      tmp_s3_path=tmp_s3_path, final_path=dest_url))
+                for source in ToastConfig().config['sources']:
+                    command_file.write('{source}\n'.format(
+                        source=json.dumps(source)))
 
             # 3. Copy command file to Hadoop filesystem
-            hadoop_tmp_command_file = mkdtemp(prefix='tmp_eggo_', dir='/tmp')
-            copy_cmd = '{hadoop_home}/bin/hadoop fs -put {source} {target}'.format(
-                hadoop_home=os.environ['HADOOP_HOME'], source=tmp_command_file,
-                target=hadoop_tmp_command_file)
-            p = Popen(copy_cmd, shell=True)
-            p.wait()
-
-            # 4. Run streaming job to download files in parallel
-            streaming_cmd = '{hadoop_home}/bin/hadoop jar {streaming_jar}' \
-                            '  -D mapred.reduce.tasks=0' \
-                            '  -D mapred.map.tasks.speculative.execution=false' \
-                            '  -D mapred.task.timeout=12000000' \
-                            '  -input {input}' \
-                            '  -inputformat org.apache.hadoop.mapred.lib.NLineInputFormat' \
-                            '  -output {output}' \
-                            '  -outputformat org.apache.hadoop.mapred.lib.NullOutputFormat' \
-                            '  -mapper bin/download_upload_mapper.sh' \
-                            '  -file bin/download_upload_mapper.sh'.format(
-                hadoop_home=os.environ['HADOOP_HOME'],
-                streaming_jar=os.environ['STREAMING_JAR'],
-                input=hadoop_tmp_command_file, output=self.destination.replace('s3:', 's3n:'))
-            p = Popen(streaming_cmd, shell=True)
-            p.wait()
-
+            hdfs_client = HdfsClient()
+            hdfs_client.put(local_path, hdfs_path)
         except:
             raise
         finally:
             rmtree(tmp_dir)
 
     def output(self):
+        return HdfsTarget(path=self.hdfs_path)
+
+
+class DownloadDatasetHadoopTask(JobTask):
+    destination = Parameter()  # full S3 prefix to put data
+
+    def requires(self):
+        # TODO: get a proper temp file here
+        tmp_hdfs_path = 'hdfs://{nn}:9000/tmp/{rand_id}'.format(
+            nn=os.environ['SPARK_MASTER'],
```

Comment:
The scheme should be omitted here so that it runs locally and uses the local file system. When run on EC2 this will use  HDFS even if the hdfs scheme is not specified, since HDFS is configured as the default file system.